### PR TITLE
Minor fixes for references

### PR DESCRIPTION
**Pull request by pangratz**

### CODE REVIEW

File: packages/ember-data/lib/system/reference.js [hunk 1]
```diff
@@ -117,7 +117,10 @@ Reference.prototype = {
     this._attributes = Ember.create(null);
 
     return PromiseObject.create({
-      promise: resolver.promise
+      promise: resolver.promise.then(() => {
```
Comment (pangratz) [inline packages/ember-data/lib/system/reference.js]: :raised_hands: thanks @fivetanley for #3092 :raised_hands:

File: packages/ember-data/lib/system/model/model.js [hunk 2]
```diff
@@ -683,10 +683,7 @@ var Model = Ember.Object.extend(Ember.Evented, {
     successfully or rejected if the adapter returns with an error.
   */
   save: function() {
-    var model = this;
-    return this.reference.save().then(function() {
-      return model;
-    });
+    return this.reference.save();
```
Comment (igorT) [inline packages/ember-data/lib/system/model/model.js]: reload wraps in the record file, so if we want to be consistent, maybe makes more sense to wrap here? Reference deals with internals, and model.js deals with exposing itself to the world?

Comment (pangratz) [inline packages/ember-data/lib/system/model/model.js]: Sounds very reasonable. This also means bye bye PromiseObject from `system/reference.js` :+1:

**Comment by igorT** [comment]
Thanks!

File: packages/ember-data/lib/system/reference.js [hunk 3]
```diff
@@ -194,7 +191,7 @@ Reference.prototype = {
     });
   },
 
-  didIinitalizeData: function() {
+  didInitalizeData: function() {
```
Comment (sly7-7) [inline packages/ember-data/lib/system/reference.js]: seems like a typo is still here

**Comment by pangratz** [comment]
lol, fixing typos, how do you even work? Good catch @sly7-7 :disappointed:

File: packages/ember-data/lib/system/reference.js (hunk 3 above)
Comment (pangratz) [inline packages/ember-data/lib/system/reference.js]: #3107 thanks again :wink:

**Pull request merged**